src/ng-select/ng-dropdown-panel.component.ts

```diff
@@ -60,6 +60,7 @@ export class NgDropdownPanelComponent implements OnInit, OnChanges, OnDestroy, A
     @Input() virtualScroll = false;
     @Input() headerTemplate: TemplateRef<any>;
     @Input() footerTemplate: TemplateRef<any>;
+    @Input() bottomDistance = 0;
```

Comment:
Thanks for your feedback @varnastadeus , it makes total sense!

Reply:
So one way of doing it would be to calculate the height of the items, and we also need to consider there might be a header and/or footer that takes up some space.

I am a bit concerned about calling `getBoundingClientRect` all the time, maybe those values can be allowed to be cached?

```javascript
private _fireScrollToEnd() {
    if (this._scrollToEndFired) {
        return;
    }
    const scroll: HTMLElement = this.scrollElementRef.nativeElement;
    const panel: HTMLElement = this._elementRef.nativeElement;
    const padding: HTMLElement = this.virtualScroll ?
        this.paddingElementRef.nativeElement :
        this.contentElementRef.nativeElement;
    
    // Refactor the breakpoint calculation into separate method
    const breakPoint = this._getTriggerBreakPoint(padding);

    if (scroll.scrollTop + panel.clientHeight >= breakPoint) {
        this.scrollToEnd.emit();
        this._scrollToEndFired = true;
    }
}

private _getTriggerBreakPoint(padding: HTMLElement) {
    // Need to consider potential header and/or footer elements being present
    const dropdownPanel: HTMLElement = this._elementRef.nativeElement;
    const additionalElementsHeight = ['.ng-dropdown-header', '.ng-dropdown-footer'].reduce((totalHeight, selector) => {
        const el = dropdownPanel.querySelector(selector);
        totalHeight += (el && el.getBoundingClientRect().height || 0);
        return totalHeight;
    }, 0);

    const { childHeight } = this._calculateDimensions();
    return padding.clientHeight + additionalElementsHeight - (this.bottomDistance * childHeight);
}
```

Reply:
And I'm all for changing the name from `bottomDistance` to something more describing